subprojects/core/src/integTest/groovy/org/gradle/api/file/FileCollectionSymlinkIntegrationTest.groovy

```diff
@@ -174,6 +174,57 @@ class FileCollectionSymlinkIntegrationTest extends AbstractIntegrationSpec {
         outputDirectory.list() == [input.name]
     }
 
+    @Issue('https://github.com/gradle/gradle/issues/9904')
+    def "task with broken symlink in inputs are valid"() {
+        def inputDirectoryWithBrokenLink = file('inputDirectoryWithBrokenLink').createDir()
+        def inputFileTarget = file("brokenInputFileTarget")
+        def inputFile = inputDirectoryWithBrokenLink.file('brokenInputFile').createLink(inputFileTarget)
+        def output = file("output.txt").createFile()
```

Comment:
Let's not create the output file before the task executed. The task should create the output file.

Reply:
Done